roles/slurm/tasks/login-compute-setup.yml

```diff
@@ -0,0 +1,32 @@
+---
+
+- name: Hide GPUs for regular user logins via sshd.service.
+  shell: |
+    deviceprop=$(systemctl show sshd.service -p DeviceAllow | grep -i nvidiactl)
+
+    if [ -z "$deviceprop" ] ; then
+      systemctl set-property sshd.service DeviceAllow="/dev/nvidiactl"
+    fi
+  args:
+    creates: "{{ '/etc/systemd/system.control/sshd.service.d/50-DeviceAllow.conf' \
+                 if ansible_os_family == 'RedHat' else \
+                 '/etc/systemd/system.control/ssh.service.d/50-DeviceAllow.conf' }}"
+  when: is_controller
+  tags:
+    - config
+
+- name: Patch lastuserjob epilog.
```

Comment:
It seems to me that the users in `/etc/slurm/localusers.backup` should always be allowed to log in anyway. 😄 Why not just modify the prolog script provided by DeepOps, instead of patching it on the fly for this use case?

Reply:
Unless I patch epilog `40-lastuserjob-processes` for the caveat of `localusers.backup` all the processes on the node for the user get killed. The user's session gets killed as well so they are logged out of all their "login" sessions not just slurm reservation session.

https://github.com/NVIDIA/deepops/blob/552f7780c878ef4403f3cfbd38ee51d7cae1f262/roles/slurm/templates/etc/slurm/epilog.d/40-lastuserjob-processes#L4-L8

Reply:
Got it. I'd still prefer we modify the file copied in by Ansible, rather than patching it in place, but I don't think that should be a blocker for the PR.

Reply:
In the general case where you might have multinode setup, you don't want to patch on compute nodes that are not used as login nodes. So I patch after the epilog is copied. The "playbooks/slurm-cluster/slurm.yml" has a specific order.

1. The epilog files are not copied yet.
```
# Set up slurm controller
- hosts: slurm-master
. . .
        is_controller: yes
```

2. Now the epilog files are copied to all the compute nodes.
```
# Set up slurm nodes
- hosts: slurm-node
. . .
        is_compute: yes
```

3. If a compute node was set as a login node it will have the epilog files after step 2 above. Now patch the epilog files for login node(s). 
```
# Set up slurm-login-on-compute if enabled
- hosts: slurm-master
. . .
        is_controller: yes
      when: slurm_login_on_compute | default(false)
```